client/starwhale/api/_impl/data_store.py

```diff
@@ -123,22 +220,136 @@ def __str__(self) -> str:
         else:
             return self.name.upper()
 
-    __repr__ = __str__
+
+class SwCompositeType(SwType):
+    def __init__(self, name: str) -> None:
+        super().__init__(name, pa.binary())
+
+    def serialize(self, value: Any) -> Any:
+        return pickle.dumps(value)
+
+    def deserialize(self, value: Any) -> Any:
+        return pickle.loads(value)
 
 
-UNKNOWN = SwType("unknown", None, 1, None)
-INT8 = SwType("int", pa.int8(), 8, 0)
-INT16 = SwType("int", pa.int16(), 16, 0)
-INT32 = SwType("int", pa.int32(), 32, 0)
-INT64 = SwType("int", pa.int64(), 64, 0)
-FLOAT16 = SwType("float", pa.float16(), 16, 0.0)
-FLOAT32 = SwType("float", pa.float32(), 32, 0.0)
-FLOAT64 = SwType("float", pa.float64(), 64, 0.0)
-BOOL = SwType("bool", pa.bool_(), 1, 0)
-STRING = SwType("string", pa.string(), 32, "")
-BYTES = SwType("bytes", pa.binary(), 32, b"")
+class SwListType(SwCompositeType):
+    def __init__(self, element_type: SwType) -> None:
+        super().__init__("list")
+        self.element_type = element_type
+
+    def merge(self, type: SwType) -> SwType:
+        if isinstance(type, SwListType):
+            t = self.element_type.merge(type.element_type)
+            if t is self.element_type:
+                return self
+            if t is type.element_type:
+                return type
+            return SwListType(t)
+        raise RuntimeError(f"conflicting type {str(self)} and {str(type)}")
```

Comment:
`raise RuntimeError(f"conflicting type {str(self)} and {str(type)}")` -> `raise RuntimeError(f"conflicting type {self} and {type}")` ?
str fmt will auto call __str__ method.

Reply:
Done